### PR TITLE
[FLINK-15327][runtime] No warning of InterruptedException during cancel.

### PR DESCRIPTION
**Pull request by AHeise**
## What is the purpose of the change

InterruptedException are previously only handled when wrapped in
WrappingRuntimeException. This patch looks through the whole exception
chain.

## Brief change log

 - Look for InterruptedException in whole exception in StreamTask.
 - Reenable previously flaky nightly tests.

## Verifying this change

- Reenabled flaky nightly tests.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / no**)**
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (**not applicable** / docs / JavaDocs / not documented)

### CODE REVIEW

File: flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/StreamTask.java [hunk 1]
```diff
@@ -484,27 +484,15 @@ public final void invoke() throws Exception {
...
-		catch (InterruptedException e) {
-			if (!canceled) {
-				Thread.currentThread().interrupt();
-				throw e;
-			}
+			mailboxProcessor.runMailboxLoop();
 		}
 		catch (Exception e) {
-			if (canceled) {
+			if (ExceptionUtils.findThrowable(e, InterruptedException.class).isPresent()) {
```
Comment (pnowojski) [inline flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/StreamTask.java]: We could add a simple test coverage for this, for example by extending `SourceStreamTaskTest#testInterruptedNotSwallowed` with a boolean parameter, controlling whether the exception is thrown wrapped or not (similar pattern as  `SourceStreamTaskTest#testCancellationWithSourceBlockedOnLock(boolean, boolean)`.

**Comment by AHeise** [comment]
@flinkbot run travis

**Comment by pnowojski** [comment]
@AHeise there are some test failures, are they related?

**Comment by AHeise** [comment]
@pnowojski Unfortunately, [one scala tests fails ](https://api.travis-ci.com/v3/job/273264866/log.txt)for real. It's unrelated to this fix though, but I could add a hotfix for it here.

**Comment by pnowojski** [comment]
It would be probably better if you created a separate jira ticket for that scala fix, so someone doesn't duplicate the effort.

Can you ping me once travis is green?

**Comment by AHeise** [comment]
@flinkbot run travis

**Pull request merged**